libcxx/test/std/ranges/range.adaptors/range.transform/general.pass.cpp

```diff
@@ -108,5 +108,22 @@ int main(int, char**) {
   }
 #endif
 
+  // GH issue #70506
```

Comment:
It looks like the following views have the same problem at a high level, because they use `__movable_box` as well:

```
take_while_view
filter_view
single_view
drop_while_view
repeat_view
transform_view
chunk_by_view
```

I think it would be worthwhile to add a regression test for each of them. Maybe a single test that checks all of them is the best approach.